packager/mpd/base/period.cc

```diff
@@ -117,6 +117,18 @@ AdaptationSet* Period::GetOrCreateAdaptationSet(
   AdaptationSet* adaptation_set_ptr = new_adaptation_set.get();
   adaptation_sets.push_back(adaptation_set_ptr);
   adaptation_sets_.emplace_back(std::move(new_adaptation_set));
+
+  if (!trickplay_cache.empty()) {
+    if (trickplay_cache.find(key) != trickplay_cache.end()) {
+      std::list<AdaptationSet*> adaptation_set_list = trickplay_cache[key];
+      for (AdaptationSet* adaptation_set : adaptation_set_list) {
+        adaptation_set->AddTrickPlayReference(adaptation_set_ptr);
```

Comment:
Need to check whether content protection matches too as in line 249. 

A few suggestions:

1. I think we can move the code to around line 222 which is better for readability, so something like:

```c++
if (media_info.video_info().has_playback_rate()) {
  // Cache trick play adaptation set if there is any.
  ...
} else {
  // Check if the adaptation set matches with any of the cached trick play adaptation set.
  ...
}
```

2. I think we can reuse FindOriginalAdaptationSetForTrickPlay here, so that we can use the function 
    - To find the original AdaptationSet of the Trick Play AdaptationSet
    - To find the trick play AdaptationSet associated with the current non-trick-play AdaptationSet
    
    The function name needs to be renamed, the result code will be something like this:

```c++
const AdaptationSet* Period::FindMatchingAdaptationSetForTrickPlay(
    const MediaInfo& media_info,
    bool content_protection_in_adaptation_set) {
  std::list<AdaptationSet*>* adaptation_sets = nullptr;
  const bool is_trickplay_adaptation_set = media_info.video_info().has_playback_rate();
  if (is_trickplay_adaptation_set) {
    MediaInfo media_info_no_trickplay = media_info;
    media_info_no_trickplay.mutable_video_info()->clear_playback_rate();
    std::string key = GetAdaptationSetKey(
        media_info_no_trickplay, mpd_options_.mpd_params.allow_codec_switching);
    adaptation_sets = &adaptation_set_list_map_[key];
  } else {
    std::string key = GetAdaptationSetKey(
        media_info, mpd_options_.mpd_params.allow_codec_switching);
    if (trickplay_cache.find(key) == trickplay_cache.end()) 
      return nullptr;
    adaptation_sets = &trickplay_cache[key];
  }
  for (AdaptationSet* adaptation_set : *adaptation_sets) {
    if (protected_adaptation_set_map_.Match(
            *adaptation_set, media_info,
            content_protection_in_adaptation_set)) {
      if (!is_trickplay_adaptation_set)
        trickplay_cache.erase(key);
      return adaptation_set;
    }
  }
  return nullptr;
}
```